NPE in PreferencesController

Probably caused by a limited set of connection profiles available.

```
2023-03-27 09:50:16,949 [main] ERROR org.rococoa.callback - Exception calling method for selector setDefaultBucketLocationGoogleStorage:
java.lang.reflect.InvocationTargetException: null
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]
	at org.rococoa.internal.OCInvocationCallbacks.callMethod(OCInvocationCallbacks.java:170) [rococoa-core-0.9.1.jar:?]
	at org.rococoa.internal.OCInvocationCallbacks.access$200(OCInvocationCallbacks.java:52) [rococoa-core-0.9.1.jar:?]
	at org.rococoa.internal.OCInvocationCallbacks$2.callback(OCInvocationCallbacks.java:82) [rococoa-core-0.9.1.jar:?]
	at jdk.internal.reflect.GeneratedMethodAccessor108.invoke(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]
	at com.sun.jna.CallbackReference$DefaultCallbackProxy.invokeCallback(CallbackReference.java:585) [jna-min-5.12.1.jar:5.12.1 (b0)]
	at com.sun.jna.CallbackReference$DefaultCallbackProxy.callback(CallbackReference.java:616) [jna-min-5.12.1.jar:5.12.1 (b0)]
	at com.sun.jna.Native.invokeInt(Native Method) ~[jna-min-5.12.1.jar:5.12.1 (b0)]
	at com.sun.jna.Function.invoke(Function.java:420) [jna-min-5.12.1.jar:5.12.1 (b0)]
	at com.sun.jna.Function.invoke(Function.java:361) [jna-min-5.12.1.jar:5.12.1 (b0)]
	at com.sun.jna.Function.invoke(Function.java:315) [jna-min-5.12.1.jar:5.12.1 (b0)]
	at org.rococoa.internal.MsgSendHandler.invoke(MsgSendHandler.java:114) [rococoa-core-0.9.1.jar:?]
	at com.sun.jna.Library$Handler.invoke(Library.java:263) [jna-min-5.12.1.jar:5.12.1 (b0)]
	at jdk.proxy2.$Proxy5.syntheticSendMessage(Unknown Source) [?:?]
	at org.rococoa.Foundation.send(Foundation.java:207) [rococoa-core-0.9.1.jar:?]
	at org.rococoa.Foundation.send(Foundation.java:181) [rococoa-core-0.9.1.jar:?]
	at org.rococoa.internal.ObjCObjectInvocationHandler.sendOnThisOrMainThread(ObjCObjectInvocationHandler.java:245) [rococoa-core-0.9.1.jar:?]
	at org.rococoa.internal.ObjCObjectInvocationHandler.invokeCocoa(ObjCObjectInvocationHandler.java:212) [rococoa-core-0.9.1.jar:?]
	at org.rococoa.internal.ObjCObjectInvocationHandler.invoke(ObjCObjectInvocationHandler.java:152) [rococoa-core-0.9.1.jar:?]
	at jdk.proxy2.$Proxy10.loadNibNamed_owner(Unknown Source) [?:?]
	at ch.cyberduck.binding.foundation.NSBundle.loadNibNamed(NSBundle.java:43) [binding-8.5.7.jar:06fe2eb1cfdb16e056b248c4e01ca0877e0a034a]
	at ch.cyberduck.binding.BundleController.loadBundle(BundleController.java:123) [binding-8.5.7.jar:06fe2eb1cfdb16e056b248c4e01ca0877e0a034a]
	at ch.cyberduck.binding.BundleController.loadBundle(BundleController.java:111) [binding-8.5.7.jar:06fe2eb1cfdb16e056b248c4e01ca0877e0a034a]
	at ch.iterate.mountainduck.application.StatusBarPreferencesController.loadBundle(StatusBarPreferencesController.java:51) [osx-4.13.6.jar:ec7d55be14969be1fec725f9a78fdb2cfd74f764]
	at ch.iterate.mountainduck.application.StatusBarPreferencesController$Factory.instance(StatusBarPreferencesController.java:127) [osx-4.13.6.jar:ec7d55be14969be1fec725f9a78fdb2cfd74f764]
	at ch.iterate.mountainduck.application.menu.StatusBarBookmarkMenuDelegate.preferencesMenuClicked(StatusBarBookmarkMenuDelegate.java:813) [osx-4.13.6.jar:ec7d55be14969be1fec725f9a78fdb2cfd74f764]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]
	at org.rococoa.internal.OCInvocationCallbacks.callMethod(OCInvocationCallbacks.java:170) [rococoa-core-0.9.1.jar:?]
	at org.rococoa.internal.OCInvocationCallbacks.access$200(OCInvocationCallbacks.java:52) [rococoa-core-0.9.1.jar:?]
	at org.rococoa.internal.OCInvocationCallbacks$2.callback(OCInvocationCallbacks.java:82) [rococoa-core-0.9.1.jar:?]
	at jdk.internal.reflect.GeneratedMethodAccessor29.invoke(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]
	at com.sun.jna.CallbackReference$DefaultCallbackProxy.invokeCallback(CallbackReference.java:585) [jna-min-5.12.1.jar:5.12.1 (b0)]
	at com.sun.jna.CallbackReference$DefaultCallbackProxy.callback(CallbackReference.java:616) [jna-min-5.12.1.jar:5.12.1 (b0)]
	at com.sun.jna.Native.invokeVoid(Native Method) ~[jna-min-5.12.1.jar:5.12.1 (b0)]
	at com.sun.jna.Function.invoke(Function.java:415) [jna-min-5.12.1.jar:5.12.1 (b0)]
	at com.sun.jna.Function.invoke(Function.java:361) [jna-min-5.12.1.jar:5.12.1 (b0)]
	at com.sun.jna.Function.invoke(Function.java:315) [jna-min-5.12.1.jar:5.12.1 (b0)]
	at org.rococoa.internal.MsgSendHandler.invoke(MsgSendHandler.java:114) [rococoa-core-0.9.1.jar:?]
	at com.sun.jna.Library$Handler.invoke(Library.java:263) [jna-min-5.12.1.jar:5.12.1 (b0)]
	at jdk.proxy2.$Proxy5.syntheticSendMessage(Unknown Source) [?:?]
	at org.rococoa.Foundation.send(Foundation.java:207) [rococoa-core-0.9.1.jar:?]
	at org.rococoa.Foundation.send(Foundation.java:181) [rococoa-core-0.9.1.jar:?]
	at org.rococoa.internal.ObjCObjectInvocationHandler.sendOnThisOrMainThread(ObjCObjectInvocationHandler.java:245) [rococoa-core-0.9.1.jar:?]
	at org.rococoa.internal.ObjCObjectInvocationHandler.invokeCocoa(ObjCObjectInvocationHandler.java:212) [rococoa-core-0.9.1.jar:?]
	at org.rococoa.internal.ObjCObjectInvocationHandler.intercept(ObjCObjectInvocationHandler.java:171) [rococoa-core-0.9.1.jar:?]
	at ch.cyberduck.binding.application.NSApplication$$ByRococoa.run(<generated>) [binding-8.5.7.jar:06fe2eb1cfdb16e056b248c4e01ca0877e0a034a]
	at ch.iterate.mountainduck.application.StatusBarApplication.main(StatusBarApplication.java:256) [osx-4.13.6.jar:ec7d55be14969be1fec725f9a78fdb2cfd74f764]
Caused by: java.lang.NullPointerException: Cannot invoke "ch.cyberduck.core.Protocol.getRegions()" because the return value of "ch.cyberduck.core.ProtocolFactory.forType(ch.cyberduck.core.Protocol$Type)" is null
	at ch.cyberduck.ui.cocoa.controller.PreferencesController.setDefaultBucketLocationGoogleStorage(PreferencesController.java:2276) ~[osx-8.5.7.jar:06fe2eb1cfdb16e056b248c4e01ca0877e0a034a]
	... 57 more
```

More locations affected:
```
Caused by: java.lang.NullPointerException: Cannot invoke "ch.cyberduck.core.Protocol.getProperties()" because the return value of "ch.cyberduck.core.ProtocolFactory.forType(ch.cyberduck.core.Protocol$Type)" is null
	at ch.cyberduck.ui.cocoa.controller.PreferencesController.setDefaultStorageClassPopup(PreferencesController.java:2200) ~[osx-8.5.7.jar:06fe2eb1cfdb16e056b248c4e01ca0877e0a034a]
	... 57 more
```

```
Caused by: java.lang.NullPointerException: Cannot invoke "ch.cyberduck.core.Protocol.getRegions()" because the return value of "ch.cyberduck.core.ProtocolFactory.forType(ch.cyberduck.core.Protocol$Type)" is null
	at ch.cyberduck.ui.cocoa.controller.PreferencesController.setDefaultBucketLocation(PreferencesController.java:2178) ~[osx-8.5.7.jar:06fe2eb1cfdb16e056b248c4e01ca0877e0a034a]
	... 57 more
```

--- a/osx/src/main/java/ch/cyberduck/ui/cocoa/controller/PreferencesController.java
+++ b/osx/src/main/java/ch/cyberduck/ui/cocoa/controller/PreferencesController.java
@@ -73,6 +73,7 @@ import java.util.Comparator;
 import java.util.EnumSet;
 import java.util.LinkedHashMap;
 import java.util.Map;
+import java.util.Set;
 import java.util.StringTokenizer;
 import java.util.concurrent.Callable;
 import java.util.concurrent.Executors;
@@ -2175,7 +2176,9 @@ public class PreferencesController extends ToolbarWindowController {
         this.defaultBucketLocation = b;
         this.defaultBucketLocation.setAutoenablesItems(false);
         this.defaultBucketLocation.removeAllItems();
-        ProtocolFactory.get().forType(Protocol.Type.s3).getRegions().stream().sorted(Comparator.comparing(Location.Name::toString)).forEach(location -> {
+        final Protocol protocol = ProtocolFactory.get().forType(Protocol.Type.s3);
+        final Set<Location.Name> regions = null == protocol ? Collections.emptySet() : protocol.getRegions();
+        regions.stream().sorted(Comparator.comparing(Location.Name::toString)).forEach(location -> {
             this.defaultBucketLocation.addItemWithTitle(location.toString());
             this.defaultBucketLocation.lastItem().setRepresentedObject(location.getIdentifier());
         });
@@ -2197,7 +2200,9 @@ public class PreferencesController extends ToolbarWindowController {
         this.defaultStorageClassPopup = b;
         this.defaultStorageClassPopup.setAutoenablesItems(false);
         this.defaultStorageClassPopup.removeAllItems();
-        for(String s : PreferencesReader.toList(ProtocolFactory.get().forType(Protocol.Type.s3).getProperties().get("s3.storage.class.options"))) {
+        final Protocol protocol = ProtocolFactory.get().forType(Protocol.Type.s3);
+        final Map<String, String> properties = null == protocol ? Collections.emptyMap() : protocol.getProperties();
+        for(String s : PreferencesReader.toList(properties.get("s3.storage.class.options"))) {
             this.defaultStorageClassPopup.addItemWithTitle(LocaleFactory.localizedString(s, "S3"));
             this.defaultStorageClassPopup.lastItem().setRepresentedObject(s);
         }
@@ -2273,7 +2278,9 @@ public class PreferencesController extends ToolbarWindowController {
         this.defaultBucketLocationGoogleStorage = b;
         this.defaultBucketLocationGoogleStorage.setAutoenablesItems(false);
         this.defaultBucketLocationGoogleStorage.removeAllItems();
-        ProtocolFactory.get().forType(Protocol.Type.googlestorage).getRegions().stream().sorted(Comparator.comparing(Location.Name::toString)).forEach(location -> {
+        final Protocol protocol = ProtocolFactory.get().forType(Protocol.Type.googlestorage);
+        final Set<Location.Name> regions = null == protocol ? Collections.emptySet() : protocol.getRegions();
+        regions.stream().sorted(Comparator.comparing(Location.Name::toString)).forEach(location -> {
             this.defaultBucketLocationGoogleStorage.addItemWithTitle(location.toString());
             this.defaultBucketLocationGoogleStorage.lastItem().setRepresentedObject(location.getIdentifier());
         });

